pkg/autoscaler/autoscaler/tidb_autoscaler.go

```diff
@@ -24,38 +24,59 @@ import (
 
 func (am *autoScalerManager) syncTiDB(tc *v1alpha1.TidbCluster, tac *v1alpha1.TidbClusterAutoScaler, client promClient.Client) error {
 	if tac.Spec.TiDB == nil {
+		emptyConsecutiveCount(tc, v1alpha1.TiDBMemberType)
 		return nil
 	}
 	sts, err := am.stsLister.StatefulSets(tc.Namespace).Get(operatorUtils.GetStatefulSetName(tc, v1alpha1.TiDBMemberType))
 	if err != nil {
 		return err
 	}
 	if !checkAutoScalingPrerequisites(tc, sts, v1alpha1.TiDBMemberType) {
+		emptyConsecutiveCount(tc, v1alpha1.TiDBMemberType)
```

Comment:
Why empty here? Even if the prerequisites are not met, the resource utilization result is correct.

Reply:
If the prerequisites are not met, the recorded count should be emptied or it can't b called as `consecutive threshold`

Reply:
`consecutive` means resource usage crosses the threshold continuously, why we empty the count for resource usage just because of upgrading or scaling?